### ISSUE TITLE
[BUG] {"detail":"Too many requests in 1 hour. Try again later."}

### ISSUE THREAD

**Issue by acheong08** [acheong08/ChatGPT#364]
This is because your IP address has been blacklisted by OpenAI. To solve this, use a VPN (proxy not recommended).

Email me if you need a free Wireguard VPN. (From a third party. The extra step is to prevent overloading their services. It may not stay free.)

**Comment by FRCassarino** [comment]
For me, I initialize the Chatbot, and the first prompt gets a response as expected, but the second one fails with that error:

`chatbot = Chatbot(config)
      chatbot.get_chat_response("This is prompt 1", output="text")
      chatbot.get_chat_response("This is prompt 2", output="text") # this one fails`

**Comment by acheong08** [comment]
Yes. It means your IP address has been blacklisted. You can check out some working VPNs at `https://vpn.fail`

**Comment by acheong08** [comment]
It seems you are still using the old version of the repository

**Comment by FRCassarino** [comment]
> It seems you are still using the old version of the repository. Please upgrade with `pip3 install --upgrade revChatGPT`

Ah I had tried to upgrade but I guess something hadn't worked. Switched to the new ask() method and it works fine now.

**Comment by DiamondDemon669** [comment]
this could also be because of an expired session token

**Comment by pepi99** [comment]
This is normal, I think. I was using the GUI of ChatGPT for a purpose when I overloaded it with many requests of very similar type. After say 40 or 50 of them, it just told me to wait 1 hour.

**Comment by acheong08** [comment]
Yup. They're trying to limit costs. You can work around it by cycling multiple accounts with the API

**Comment by sbaliyun** [comment]
try use https://chatgpt.sbaliyun.com/

**Comment by acheong08** [comment]
@sbaliyun It's very slow compared to the OpenAI API

**Comment by acheong08** [comment]
And it's not even ChatGPT. That is just GPT-3

**Comment by acheong08** [comment]
Check out [acheong08/ChatGPT-lite](https://github.com/acheong08/ChatGPT-lite). It solves all the annoying browser problems at the cost of your privacy

**Comment by steeljardas** [comment]
@acheong08 Have you added any built-in delay between requests to avoid people getting rate limited? If not do you have any idea how much time we should wait between request? Also does ChatGPT-lite solve the issue of getting rate limited?

**Comment by AprilNEA** [comment]
Does anyone have statistics on what this rate limit is approximately ?

**Comment by acheong08** [comment]
> @acheong08 Have you added any built-in delay between requests to avoid people getting rate limited? If not do you have any idea how much time we should wait between request? Also does ChatGPT-lite solve the issue of getting rate limited?

It solves IP based rate limiting but not account rate limits. There are no built in rate limit preventions yet

**Comment by acheong08** [comment]
> Does anyone have statistics on what this rate limit is approximately ?

I'm getting rate limited after ~50 requests

**Comment by Andy7737** [comment]
This is a CLI issue and not an app issue, but pasting in to cmd with line breaks causes it to enter each new sentence after the other, creating a bunch of requests unnecessarily. I don't know if !rollback helps with this?

Anyway, tried using NordVPN and refreshing session, but I still get "Too many requests". If I where to switch to a new session-token wouldn't that create a new conversation, forcing you to begin from scratch with your prompts?

**Comment by wise-introvert** [comment]
How long does the rate-limit last? an hour?

**Comment by Piyush-kumarr** [comment]
it this message show than change your account or make another account with other email then you have two chat gpt if one stop then another

**Comment by dbbice** [comment]
> it this message show than change your account or make another account with other email then you have two chat gpt if one stop then another

does this method work？How many requests to run in ten seconds?

**Comment by dbbice** [comment]
> Can someone please tell me how long the rate-limit lasts? an hour?
> 
> TIA 😄

Have you solved it now?How many requests to run in ten seconds?

**Comment by dbbice** [comment]
> Can someone please tell me how long the rate-limit lasts? an hour?
> 
> TIA 😄

Have you solved it now?How many requests to run in ten seconds?

**Comment by acheong08** [comment]
5 requests per 10 seconds for the proxy. There is different rate limit from OpenAI based on accounts but you can cycle them.
